controller/src/main/java/io/pravega/controller/server/impl/ControllerServiceConfigImpl.java

```diff
@@ -38,6 +38,8 @@
     private final boolean controllerClusterListenerEnabled;
     private final TimeoutServiceConfig timeoutServiceConfig;
 
+    private final String tlsEnabledForSegmentStore;
```

Comment:
why is this a string and not a boolean?

Reply:
The value of this field can be true/false/empty. If we make this a boolean, it'll need to be either true/false. I could make it `Optional<Boolean>`, but using `Optional<>` as a method argument is considered as anti-pattern by many.